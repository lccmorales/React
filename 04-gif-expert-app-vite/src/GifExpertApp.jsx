import { useState } from "react";
import { AddForm, GifGrid } from './components';

export const GifExpertApp = ( { defaultCategories = []} ) => {
    const [ categories, setCategories ] = useState( defaultCategories );
    const [ limitNumGifs, setLimitNumGifs ] = useState(2);
    
    const onAddForm = ( newCategory, newLimitNumGif ) => {
        console.log(newCategory, ' - ', newLimitNumGif);
        if ( categories.includes(newCategory) )  return;
        setCategories([ newCategory, ...categories ]);
        setLimitNumGifs( parseInt(newLimitNumGif) );
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            {/* <AddForm setCategories={ setCategories } setLimitNumGifs={ setLimitNumGifs }/> */}
            <AddForm onNewForm = { (cat, limit) => onAddForm(cat, limit) }/>
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid key={ category } category={ category } limitNumGifs={ limitNumGifs }/>
                    ))
                }
            </ol>
        </>
    );
}