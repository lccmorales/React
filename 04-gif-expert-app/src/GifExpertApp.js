import React, { useState } from 'react';
import { AddForm } from './components/AddForm';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {
    const [ categories, setCategories ] = useState(['Arcane', 'Demon Slayer']);
    const [ limitNumGifs, setLimitNumGifs ] = useState(2);
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddForm setCategories={ setCategories } setLimitNumGifs={ setLimitNumGifs }/>
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid key={ category } category={ category } limitNumGifs={ limitNumGifs }/>
                    ))
                }
            </ol>
        </>
    );
}
export default GifExpertApp;