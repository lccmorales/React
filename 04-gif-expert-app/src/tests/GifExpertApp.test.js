import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en GifExpertApp.js', () => {
    // Variables Globales
    // const category = 'Arcane';
    // const limitNumGifs = 2;
    
    test('debe de mostrar <GifExpertApp /> correctamente', () => {  
        const wrapper = shallow(<GifExpertApp/>);
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['Arcane', 'One Piece'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
});