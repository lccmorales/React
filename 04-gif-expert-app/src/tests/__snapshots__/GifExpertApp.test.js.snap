// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp.js debe de mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddForm
    setCategories={[Function]}
    setLimitNumGifs={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GifExpertApp.js debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddForm
    setCategories={[Function]}
    setLimitNumGifs={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Arcane"
      key="Arcane"
      limitNumGifs={2}
    />
    <GifGrid
      category="One Piece"
      key="One Piece"
      limitNumGifs={2}
    />
  </ol>
</Fragment>
`;
